src/App.jsx

```diff
@@ -14,45 +17,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const sortByDefault = {
+  method: '',
+  isReversedOrder: false,
+};
+
+const getPreparedGoods = (list, { method, isReversedOrder }) => {
+  const preparedGoods = [...list];
+
+  if (method) {
+    switch (method) {
+      case (SORT_BY_ALPHABET):
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+
+      case (SORT_BY_LENGTH):
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (isReversedOrder) {
+    preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState(sortByDefault);
+  const goods = getPreparedGoods(goodsFromServer, sortBy);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortBy({ ...sortBy, method: SORT_BY_ALPHABET })}
```

Comment:
You can write a function handler for it, not use anonim function ( not critical)